src/app/App.java

```diff
@@ -74,14 +60,65 @@ public static void main(String[] args) throws Exception {
                 while (it.hasNext()) {
                     System.out.println(it.next().getName());
                 }
+            } else if ("message add".equals(line)) {
+                if (App.topics.size() < 1) {
+                    System.out.println("No topics available. Please add topic before...");
+                    continue;
+                }
+                Topic topic = topicCliChooser();
+                System.out.println("Add a new message to topic " + topic.getName() + ":");
+                System.out.print("Messages' name: ");
+                String messageName = getLine();
+                System.out.print("Messages' content: ");
+                String messageContent = getLine(); // maybe content should be more as one line ;-)
+                Message message = new Message(topic, messageName, messageContent);
+                multicastPublisher.sendMessage(message);
+            } else if ("message print".equals(line)) {
+                if (App.topics.size() < 1) {
+                    System.out.println("No topics available. Please add topic before...");
+                    continue;
+                }
+                Topic topic = topicCliChooser();
+                System.out.println("The " + App.messages.size() + " messages to topic " + topic.getName() + " are:");
+                Iterator<Message> it = App.messages.iterator();
+                while (it.hasNext()) {
+                    Message message = it.next();
+                    if (message.getTopic().equals(topic)) {
+                        System.out.println("Message Name: " + message.getName());
+                        System.out.println("Message Content: " + message.getContent());
+                    }
+                    if (it.hasNext()) {
+                        // Print divider between messages if there are more on the list
+                        System.out.println("----------------------------------------------");
+                    }
+                }
             } else {
                 System.out.println("Did not understand that command. Use 'help' for further assistance.");
             }
         }
+
     }
 
     private static String getLine() throws IOException {
         BufferedReader buffer = new BufferedReader(new InputStreamReader(System.in));
         return buffer.readLine();
     }
+
+    private static Topic topicCliChooser() throws IOException {
+        System.out.println("To which topic you want to add an message?");
+        Iterator<Topic> it = App.topics.iterator();
+        while (it.hasNext()) {
+            Topic element = it.next();
+            int index = App.topics.indexOf(element);
+            System.out.println(index + ": " + element.getName());
+        }
```

Comment:
Hier bietet sich eine foreach-Schleife an. Z.B. so:
        int index;
        for (Topic topic : App.topics) {
            index = App.topics.indexOf(topic);
            System.out.println(index + ": " + topic.getName());            
        }

Reply:
Gibt es bis auf die kürzere Notation irgendwelche anderen Vorteile?

Reply:
Nein, es ist vor allem kürzer. Das Auslagern der index-Variable vor die Schleife ist bei beiden Schleifen-Typen sinnvoll.